src/modules/tools/zprobe/CalibrationStrategy.cpp

```diff
@@ -0,0 +1,597 @@
+}
+
+
+// fills out n actuator positions where the probe triggered
+// returns false if any probe failed
+bool CalibrationStrategy::probe_spiral(int n, int repeats, float actuator_positions[/*n*/][3]) {
+    if (THEKERNEL->robot->actuators.size() != 3) return false;
+    // Spiraling probe pattern
+
+    // Archimedes' spiral r = a * theta
+    // arc_length(x) = integrate a * theta * dtheta from 0 to x = a*x^2/2
+    // inversely, theta(len) = sqrt(2*len/a)
+    // we want outer point to be at probe_radius -> theta_max = probe_radius/a
+    // step_length = length(theta_max)) / n == probe_radius^2/(2*a*n)
+    // solve a * 2pi == step_length -> a = probe_radius / (2*sqrt(n)*sqrt(pi))
+
+    float a = probe_radius / (2 * sqrt(n * M_PI));
+
+    auto theta = [a](float length) {
+        return sqrt(2*length/a);
+    };
+    double step_length = probe_radius * probe_radius / (2 * a * n);
+
+    // move the probe to it's starting position a specified height above the bed
+    setup_probe();
+
+    for (int i = 0; i < n; i++) {
+        float angle = theta(i * step_length);
+        float r = angle * a;
+        // polar to cartesian
+        float x = r * cos(angle);
+        float y = r * sin(angle);
+
+        int steps; // dummy
+        if (!zprobe->doProbeAt(steps, x, y, actuator_positions[i], repeats)) return false;
+        // remove trim adjustment from returned actuator positions
+        for (int j = 0; j < 3; j++)
+            actuator_positions[i][j] += trim[j];
+    }
+
+    return true;
+}
+
+// fills out n actuator positions where the probe triggered
+// returns false if any probe failed
+bool CalibrationStrategy::probe_symmetric(int n, int repeats, float actuator_positions[/*n*/][3]) {
+    if (THEKERNEL->robot->actuators.size() != 3) return false;
+
+    // move the probe to it's starting position a specified height above the bed
+    setup_probe();
+
+    int position_index = 0;
+    auto probe = [&](float x, float y) {
+        int steps; // dummy
+        bool success = zprobe->doProbeAt(steps, x, y, actuator_positions[position_index], repeats);
+        // remove trim adjustment from returned actuator positions
+        for (int j = 0; j < 3; j++)
+            actuator_positions[position_index][j] += trim[j];
+        position_index++;
+        return success;
+    };
+
+    // probe the center point except for n==2 and n==3
+    if (n != 2 && n != 3) {
+        if (!probe(0,0)) return false;
+        n--;
+    }
+    for (int i = 0; i < n; i++) {
+        float angle = M_PI*2*i/n;
+
+        float x = probe_radius * cos(angle);
+        float y = probe_radius * sin(angle);
+        if (!probe(x,y)) return false;
+    }
+
+    return true;
+}
+
+
+float CalibrationStrategy::compute_model_error(float const actuator_position[3], float home_offs_z) {
+    float cartesian[3];
+    float trimmed_position[3];
+    // simulate the effect of trim
+    for (int i = 0; i < 3; i++) trimmed_position[i] = actuator_position[i] - trim[i];
```

Comment:
Actually, I remove the initial trim from the actuator positions. Thats why I can re-apply it here.